state-sync/state-sync-v1/src/request_manager.rs

```diff
@@ -206,7 +206,7 @@ impl RequestManager {
                 let weights: Vec<_> = peers
                     .iter()
                     .map(|(peer, peer_score)| {
-                        eligible_peers.push((*peer).clone());
+                        eligible_peers.push(**peer);
```

Comment:
nit: I wonder if we can avoid the double star (just for readability)?

Reply:
It would only be avoiding the star if passing references to the output (too much work), or by first dereferencing up above collecting by network_level, which would still need a * here.  So, I'd prefer 2 stars in one place vs a star in 2 places.  It's just a weird one.